app/events/zync_event.rb

```diff
@@ -19,7 +19,7 @@ def self.create(event, model = event)
     new(
       metadata: {
         provider_id: provider_id,
-      },
+      }.reverse_merge(metadata),
```

Comment:
Probably only if the event is a ZyncEvent (but not sure)
Anyway this won't do any harm